CMakeLists.txt

```diff
@@ -482,8 +482,18 @@ endif()
 if(FLB_UNICODE_ENCODER)
   enable_language(CXX)
   set (CMAKE_CXX_STANDARD 11)
-  add_subdirectory(${FLB_PATH_LIB_SIMDUTF} EXCLUDE_FROM_ALL)
-  FLB_DEFINITION(FLB_HAVE_UNICODE_ENCODER)
+  # The current minimum required gcc and clang versions are sited from
+  # https://github.com/fluent/fluent-bit/blob/master/.github/workflows/unit-tests.yaml.
+  # gcc version is 7.0 and clang version is 6.0 now.
+  if (MSVC OR
+      (CMAKE_COMPILER_IS_CLANG AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 6.0) OR
+      (CMAKE_COMPILER_IS_GNUCC AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 7.0))
+    add_subdirectory(${FLB_PATH_LIB_SIMDUTF} EXCLUDE_FROM_ALL)
+    FLB_DEFINITION(FLB_HAVE_UNICODE_ENCODER)
+  else()
+    message(STATUS "${CMAKE_CXX_COMPILER} version: ${CMAKE_CXX_COMPILER_VERSION} is too old. Disabling simdutf library build.")
```

Comment:
Probably should be a warning level.

I'm not a fan of implicitly disabling things when people may have request it to be enabled.

Separately I raised that if we explicitly turn something on then we really should fail if we turn it off.